spec/mem/pointer.md

```diff
@@ -57,11 +86,16 @@ pub enum PtrType {
         mutbl: Mutability,
         /// Describes what we know about the pointee.
         pointee: PointeeInfo,
+        /// Indicates what kind of metadata this pointer carries.
+        meta_kind: PointerMetaKind,
```

Comment:
Wouldn't it make sense to put this inside `PointeeInfo`?

Reply:
Maybe. But it felt a bit nicer to have all meta_kind here and not just in the raw variant. And I didn't want to mess with the PointeeInfo if not necessary, since the other PR is using it aswell.

There is also a debate to be had, whether using `Option<T>` or `T` with a `None` variant is nicer. I noticed a the weird decision of mine to go for option 1 with `PointerMeta`, which makes checking and constructing `None`s easier and option 2 with `PointerMetaKind`, which avoids having to have the signature of `Option<T>` whenever it is used.

Reply:
IMO this is conceptually clearly part of `PointeeInfo` -- it tells us something about the pointee type -- so unless there's a strong argument to the contrary, please add it there. In MIR, a reference has mutability and pointee type; in MiniRust, we replace pointee type by `PointeeInfo`. But then adding a third thing here doesn't have a good justification IMO.